### PR TITLE
feat(Public API): Make `BottomControlsResource#addControls` public

### PR DESCRIPTION
**Pull request by Aunali321**
…Patch` patch public

Motivation: Allows third party patch bundles to add controls to bottom controls bar

### CODE REVIEW

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Please run the apiDump task

File: src/main/kotlin/app/revanced/patches/youtube/misc/playercontrols/BottomControlsResourcePatch.kt [hunk 1]
```diff
@@ -11,7 +11,7 @@ import java.io.Closeable
 object BottomControlsResourcePatch : ResourcePatch(), Closeable {
     internal var bottomUiContainerResourceId: Long = -1
 
-    internal const val TARGET_RESOURCE_NAME = "youtube_controls_bottom_ui_container.xml"
+    const val TARGET_RESOURCE_NAME = "youtube_controls_bottom_ui_container.xml"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/misc/playercontrols/BottomControlsResourcePatch.kt:14]: Why has this been made public?

Comment (Aunali321) [reply src/main/kotlin/app/revanced/patches/youtube/misc/playercontrols/BottomControlsResourcePatch.kt:14]: https://github.com/ReVanced/revanced-patches/blob/50d3195845488bed19c71340b9c95bb3487c5cdb/src/main/kotlin/app/revanced/patches/youtube/interaction/downloads/ExternalDownloadsResourcePatch.kt#L52

Comment (Aunali321) [reply src/main/kotlin/app/revanced/patches/youtube/misc/playercontrols/BottomControlsResourcePatch.kt:14]: Can i access `BottomControlsResourcePatch.TARGET_RESOURCE_NAME` from another patch if its internal? if so i will revert

File: src/main/kotlin/app/revanced/patches/youtube/interaction/downloads/ExternalDownloadsResourcePatch.kt [hunk 2]
```diff
@@ -28,27 +28,36 @@ internal object ExternalDownloadsResourcePatch : ResourcePatch() {
...
-        // Copy resources
-        context.copyResources("downloads", ResourceGroup("drawable", "revanced_yt_download_button.xml"))
+        context.copyResources(
+            "downloads",
+            ResourceGroup("drawable", "revanced_yt_download_button.xml")
+        )
 
-        // Add download button node
-        BottomControlsResourcePatch.addControls("downloads/host/layout/${BottomControlsResourcePatch.TARGET_RESOURCE_NAME}")
+        // Add download button node.
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/interaction/downloads/ExternalDownloadsResourcePatch.kt:60]:
```suggestion
```

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**